### PR TITLE
feat: Improve handling of deep type guards

### PR DESCRIPTION
**Pull request by sunrabbit123**
**Description:**

I resolve desctructure assigned about Interface Union

### CODE REVIEW

File: crates/stc_ts_file_analyzer/src/analyzer/expr/bin.rs [hunk 1]
```diff
@@ -2302,7 +2302,60 @@ impl Analyzer<'_, '_> {
         if let Type::Union(Union { types, .. }) = origin_ty.normalize() {
             for ty in types {
                 match ty.normalize() {
-                    Type::TypeLit(tl @ TypeLit { members, .. }) => {
+                    Type::Interface(interface) => {
```
Comment (kdy1) [inline crates/stc_ts_file_analyzer/src/analyzer/expr/bin.rs:2305]: As this logic is duplicate of the logic below, I think it's better to extract this as a method and call it

**Review by kdy1 — CHANGES_REQUESTED** [review]
Thank you!

File: crates/stc_ts_file_analyzer/src/analyzer/expr/bin.rs (hunk 1 above)
Comment (kdy1) [reply crates/stc_ts_file_analyzer/src/analyzer/expr/bin.rs:2305]: I think `get_additional_exclude_target_for_type_lit`  is the ideal name

Comment (sunrabbit123) [reply crates/stc_ts_file_analyzer/src/analyzer/expr/bin.rs:2305]: thx, i modifyed it

**Review by kdy1 — APPROVED** [review]
Thanks!

**Comment by kdy1** [comment]
Oh `cpelll` failed. Can you rename it to `type_lit` instead?

**Comment by kdy1** [comment]
It still fails with
```
/home/runner/work/stc/stc/crates/stc_ts_file_analyzer/src/analyzer/expr/bin.rs:2308:68 - Unknown word (typelit)
/home/runner/work/stc/stc/crates/stc_ts_file_analyzer/src/analyzer/expr/bin.rs:2320:64 - Unknown word (typelit)
/home/runner/work/stc/stc/crates/stc_ts_file_analyzer/src/analyzer/expr/bin.rs:2376:42 - Unknown word (typelit)

```

**Comment by sunrabbit123** [comment]
omg, but this error is from function name `get_additional_exclude_target_for_type_lit`...
oh i missed underbar

**Review by kdy1 — APPROVED** [review]

**Pull request merged**